### PR TITLE
Introduce VGS http response object

### PR DESCRIPTION
**Pull request by moehajj**

### CODE REVIEW

File: larky/src/main/resources/vgs/http/response.star [hunk 1]
```diff
@@ -0,0 +1,47 @@
+load("@stdlib//larky", larky="larky")
+load("@stdlib//urllib/parse", "parse")
```
Comment (mahmoudimus) [inline larky/src/main/resources/vgs/http/response.star:2]: `parse="parse"`

File: larky/src/main/resources/vgs/http/response.star [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+def VGSHttpResponse(
+    body=None,
+    headers={},
```
Comment (mahmoudimus) [inline larky/src/main/resources/vgs/http/response.star:6]: `headers=None`

File: larky/src/main/resources/vgs/http/response.star [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+                self.remove_header("Content-length")
+    self.body = larky.property(_get_body, _set_body)
+
+    def add_header(key, val):
+        self.headers[key] = val
+    self.add_header = add_header
+
+    def __init__(
+        body=None,
+        headers={},
```
Comment (mahmoudimus) [inline larky/src/main/resources/vgs/http/response.star:32]: no point in having these defaults be in `__init__` if they're already defined here: https://github.com/verygoodsecurity/starlarky/pull/244/files#diff-1d171479d8b2520c1c1417a2bf8e5f8f6c9fdcc448bae315827aff445b3064a6R4

File: larky/src/main/resources/vgs/http/response.star [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+        status_code=200
+    ):
+        self._body = body
+
+        self.headers = {}
+        for key, value in headers.items():
```
Comment (mahmoudimus) [inline larky/src/main/resources/vgs/http/response.star:38]:
```suggestion
        headers = headers or {}
        for key, value in headers.items():
```

**Comment by mahmoudimus** [comment]
Any change in `response.body` will require us to properly update the `content-length` header. What is going to do that for us? Are we going to ask customers to be aware here? How will you respect the `content-encoding`?

**Review by mahmoudimus — CHANGES_REQUESTED** [review]

File: larky/src/main/resources/vgs/http/request.star [hunk 5]
```diff
@@ -19,6 +19,10 @@ def VGSHttpRequest(
         return self.full_url
     def _set_url(url):
         self.full_url = url
+
+        parsed_url = parse.urlsplit(url)
+        self.path = parsed_url.path
+        self.query_string = parsed_url.query
```
Comment (aslepakurov) [inline larky/src/main/resources/vgs/http/request.star:25]: Maybe if we are separating path and query string, let's provide vars for every other part of url?

Comment (moehajj) [reply larky/src/main/resources/vgs/http/request.star:25]: `urllib` takes care of generating `type` , `host` and `fragment` already, so we extended it so fill in the gap

File: larky/src/test/resources/vgs_tests/http/test_default_response.star [hunk 6]
```diff
@@ -0,0 +1,86 @@
+"""Unit tests for request.star"""
```
Comment (mahmoudimus) [inline larky/src/test/resources/vgs_tests/http/test_default_response.star:1]:
```suggestion
"""Unit tests for response.star"""
```

File: larky/src/test/resources/vgs_tests/http/test_default_response.star [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+load("@stdlib//unittest", "unittest")
```
Comment (mahmoudimus) [inline larky/src/test/resources/vgs_tests/http/test_default_response.star:2]: Can you sort these?
And keep them consistent ? 

Use the binding: unittest="unittest"

File: larky/src/main/resources/vgs/http/response.star [hunk 8]
```diff
@@ -0,0 +1,47 @@
+load("@stdlib//larky", larky="larky")
+
+def VGSHttpResponse(
+    body=None,
+    headers=None,
+    status_code=200
+):
+
+    self = larky.mutablestruct(__name__="VGSHttpResponse", __class__=VGSHttpResponse)
```
Comment (mahmoudimus) [inline larky/src/main/resources/vgs/http/response.star:9]: spacing between equals ?

File: larky/src/main/resources/vgs/http/response.star [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+    def _get_body():
+        return self._body
+
+    def _set_body(body):
+        if body != self._body:
+            self._body = body
+            # issue 16464
+            # if we change data we need to remove content-length header
+            # (cause it's most probably calculated for previous value)
+            if "Content-length" in self.headers:
```
Comment (mahmoudimus) [inline larky/src/main/resources/vgs/http/response.star:21]: Oh we do this here!

File: larky/src/main/resources/vgs/http/response.star [hunk 10]
```diff
@@ -0,0 +1,47 @@
...
+        status_code
+    ):
+        self._body = body
+
+        headers = headers or {}
+        self.headers = {}
+        for key, value in headers.items():
+            self.add_header(key, value)
+
+        self.status_code=status_code
```
Comment (mahmoudimus) [inline larky/src/main/resources/vgs/http/response.star:41]:
```suggestion
        self.status_code = status_code
```

**Review by mahmoudimus — CHANGES_REQUESTED** [review]

**Review by aslepakurov — APPROVED** [review]

**Review by mahmoudimus — APPROVED** [review]
All good

**Pull request merged**